### ISSUE TITLE
[BUG] Controller returns OFFSET_NOT_MATCHED when commit the same object twice

### ISSUE THREAD

**Issue by Chillax-0v0** [AutoMQ/automq#1055]
controller.logs:
```
[2024-03-29 17:04:58,450] INFO [Controller 3] [CommitStreamSetObject]: nodeId=103 commit object: 1292168 success, compacted objects: [], stream range: [ObjectStreamRange(streamId=3, streamEpoch=0, startOffset=13802, endOffset=13804), ObjectStreamRange(streamId=8, streamEpoch=0, startOffset=13807, endOffset=13808), ObjectStreamRange(streamId=9, streamEpoch=0, startOffset=13822, endOffset=13823), ObjectStreamRange(streamId=37, streamEpoch=0, startOffset=13793, endOffset=13794), ObjectStreamRange(streamId=38, streamEpoch=0, startOffset=13780, endOffset=13781), ObjectStreamRange(streamId=39, streamEpoch=0, startOffset=13756, endOffset=13757), ObjectStreamRange(streamId=40, streamEpoch=0, startOffset=13769, endOffset=13770), ObjectStreamRange(streamId=41, streamEpoch=0, startOffset=13794, endOffset=13796), ObjectStreamRange(streamId=42, streamEpoch=0, startOffset=13777, endOffset=13778), ObjectStreamRange(streamId=43, streamEpoch=0, startOffset=13776, endOffset=13777), ObjectStreamRange(streamId=75, streamEpoch=0, startOffset=3103716, endOffset=3103932), ObjectStreamRange(streamId=93, streamEpoch=0, startOffset=3101424, endOffset=3101664), ObjectStreamRange(streamId=129, streamEpoch=0, startOffset=3102384, endOffset=3102612), ObjectStreamRange(streamId=136, streamEpoch=0, startOffset=3101160, endOffset=3101364), ObjectStreamRange(streamId=140, streamEpoch=0, startOffset=3106608, endOffset=3106824), ObjectStreamRange(streamId=148, streamEpoch=0, startOffset=3100716, endOffset=3100932), ObjectStreamRange(streamId=149, streamEpoch=0, startOffset=3101244, endOffset=3101460), ObjectStreamRange(streamId=150, streamEpoch=0, startOffset=3104148, endOffset=3104364), ObjectStreamRange(streamId=151, streamEpoch=0, startOffset=3101796, endOffset=3102012), ObjectStreamRange(streamId=153, streamEpoch=0, startOffset=3102816, endOffset=3103032), ObjectStreamRange(streamId=154, streamEpoch=0, startOffset=3101532, endOffset=3101760), ObjectStreamRange(streamId=170, streamEpoch=0, startOffset=3102456, endOffset=3102696), ObjectStreamRange(streamId=179, streamEpoch=0, startOffset=3101772, endOffset=3101976), ObjectStreamRange(streamId=186, streamEpoch=0, startOffset=13810, endOffset=13812), ObjectStreamRange(streamId=187, streamEpoch=0, startOffset=13804, endOffset=13806), ObjectStreamRange(streamId=189, streamEpoch=0, startOffset=13814, endOffset=13815), ObjectStreamRange(streamId=190, streamEpoch=0, startOffset=13830, endOffset=13831), ObjectStreamRange(streamId=208, streamEpoch=0, startOffset=13795, endOffset=13797), ObjectStreamRange(streamId=209, streamEpoch=0, startOffset=13787, endOffset=13791), ObjectStreamRange(streamId=212, streamEpoch=0, startOffset=13800, endOffset=13802), ObjectStreamRange(streamId=213, streamEpoch=0, startOffset=13803, endOffset=13805), ObjectStreamRange(streamId=214, streamEpoch=0, startOffset=13792, endOffset=13793), ObjectStreamRange(streamId=215, streamEpoch=0, startOffset=13793, endOffset=13794), ObjectStreamRange(streamId=238, streamEpoch=0, startOffset=13787, endOffset=13789), ObjectStreamRange(streamId=264, streamEpoch=0, startOffset=3107112, endOffset=3107340), ObjectStreamRange(streamId=287, streamEpoch=0, startOffset=3098976, endOffset=3099192), ObjectStreamRange(streamId=312, streamEpoch=0, startOffset=3099528, endOffset=3099756), ObjectStreamRange(streamId=313, streamEpoch=0, startOffset=3101952, endOffset=3102156), ObjectStreamRange(streamId=316, streamEpoch=0, startOffset=3103128, endOffset=3103416), ObjectStreamRange(streamId=326, streamEpoch=0, startOffset=3099864, endOffset=3100092), ObjectStreamRange(streamId=335, streamEpoch=0, startOffset=3102780, endOffset=3102996), ObjectStreamRange(streamId=336, streamEpoch=0, startOffset=3103632, endOffset=3103860), ObjectStreamRange(streamId=337, streamEpoch=0, startOffset=3102708, endOffset=3102948), ObjectStreamRange(streamId=338, streamEpoch=0, startOffset=3100116, endOffset=3100344), ObjectStreamRange(streamId=339, streamEpoch=0, startOffset=3100284, endOffset=3100512), ObjectStreamRange(streamId=340, streamEpoch=0, startOffset=3103548, endOffset=3103764), ObjectStreamRange(streamId=357, streamEpoch=0, startOffset=3103572, endOffset=3103788), ObjectStreamRange(streamId=361, streamEpoch=0, startOffset=3100944, endOffset=3101172)], stream objects: [StreamObject(objectId=1292169, objectSize=18845369, streamId=63, startOffset=18858143, endOffset=18859456), StreamObject(objectId=1292170, objectSize=20605421, streamId=66, startOffset=18843408, endOffset=18844838), StreamObject(objectId=1292172, objectSize=18452663, streamId=94, startOffset=18859908, endOffset=18861204), StreamObject(objectId=1292171, objectSize=19825431, streamId=69, startOffset=18849607, endOffset=18850986), StreamObject(objectId=1292173, objectSize=18827245, streamId=95, startOffset=18847409, endOffset=18848724), StreamObject(objectId=1292175, objectSize=18253912, streamId=98, startOffset=18845086, endOffset=18846344), StreamObject(objectId=1292176, objectSize=19987907, streamId=99, startOffset=18857041, endOffset=18858426), StreamObject(objectId=1292174, objectSize=19395194, streamId=97, startOffset=18845226, endOffset=18846573), StreamObject(objectId=1292177, objectSize=19926531, streamId=100, startOffset=18841493, endOffset=18842878), StreamObject(objectId=1292178, objectSize=19453587, streamId=101, startOffset=18838950, endOffset=18840305), StreamObject(objectId=1292179, objectSize=21461007, streamId=110, startOffset=18854824, endOffset=18856315), StreamObject(objectId=1292181, objectSize=19707925, streamId=119, startOffset=18873372, endOffset=18874755), StreamObject(objectId=1292180, objectSize=18869991, streamId=112, startOffset=18845500, endOffset=18846813), StreamObject(objectId=1292182, objectSize=19830046, streamId=246, startOffset=18881433, endOffset=18882810), StreamObject(objectId=1292183, objectSize=19104665, streamId=248, startOffset=18827836, endOffset=18829190), StreamObject(objectId=1292184, objectSize=18695700, streamId=249, startOffset=18848284, endOffset=18849576), StreamObject(objectId=1292186, objectSize=20067807, streamId=261, startOffset=18834267, endOffset=18835667), StreamObject(objectId=1292185, objectSize=19871556, streamId=253, startOffset=18833837, endOffset=18835234), StreamObject(objectId=1292187, objectSize=19716157, streamId=265, startOffset=18857910, endOffset=18859300), StreamObject(objectId=1292189, objectSize=19934410, streamId=268, startOffset=18837165, endOffset=18838541), StreamObject(objectId=1292192, objectSize=19402293, streamId=271, startOffset=18853192, endOffset=18854545), StreamObject(objectId=1292191, objectSize=20500248, streamId=270, startOffset=18832904, endOffset=18834333), StreamObject(objectId=1292190, objectSize=25413288, streamId=269, startOffset=18856700, endOffset=18858500), StreamObject(objectId=1292193, objectSize=18937810, streamId=272, startOffset=18860625, endOffset=18861937), StreamObject(objectId=1292194, objectSize=19596857, streamId=289, startOffset=18865377, endOffset=18866740), StreamObject(objectId=1292195, objectSize=20361519, streamId=299, startOffset=18839288, endOffset=18840706), StreamObject(objectId=1292188, objectSize=21042994, streamId=266, startOffset=18851296, endOffset=18852765)] (org.apache.kafka.controller.stream.StreamControlManager)

[2024-03-29 17:08:34,134] ERROR [Controller 3] [CommitStreamSetObject] streamId=[StreamOffsetRange(streamId=3, startOffset=13802, endOffset=13804), StreamOffsetRange(streamId=8, startOffset=13807, endOffset=13808), StreamOffsetRange(streamId=9, startOffset=13822, endOffset=13823), StreamOffsetRange(streamId=37, startOffset=13793, endOffset=13794), StreamOffsetRange(streamId=38, startOffset=13780, endOffset=13781), StreamOffsetRange(streamId=39, startOffset=13756, endOffset=13757), StreamOffsetRange(streamId=40, startOffset=13769, endOffset=13770), StreamOffsetRange(streamId=41, startOffset=13794, endOffset=13796), StreamOffsetRange(streamId=42, startOffset=13777, endOffset=13778), StreamOffsetRange(streamId=43, startOffset=13776, endOffset=13777), StreamOffsetRange(streamId=75, startOffset=3103716, endOffset=3103932), StreamOffsetRange(streamId=93, startOffset=3101424, endOffset=3101664), StreamOffsetRange(streamId=129, startOffset=3102384, endOffset=3102612), StreamOffsetRange(streamId=136, startOffset=3101160, endOffset=3101364), StreamOffsetRange(streamId=140, startOffset=3106608, endOffset=3106824), StreamOffsetRange(streamId=148, startOffset=3100716, endOffset=3100932), StreamOffsetRange(streamId=149, startOffset=3101244, endOffset=3101460), StreamOffsetRange(streamId=150, startOffset=3104148, endOffset=3104364), StreamOffsetRange(streamId=151, startOffset=3101796, endOffset=3102012), StreamOffsetRange(streamId=153, startOffset=3102816, endOffset=3103032), StreamOffsetRange(streamId=154, startOffset=3101532, endOffset=3101760), StreamOffsetRange(streamId=170, startOffset=3102456, endOffset=3102696), StreamOffsetRange(streamId=179, startOffset=3101772, endOffset=3101976), StreamOffsetRange(streamId=186, startOffset=13810, endOffset=13812), StreamOffsetRange(streamId=187, startOffset=13804, endOffset=13806), StreamOffsetRange(streamId=189, startOffset=13814, endOffset=13815), StreamOffsetRange(streamId=190, startOffset=13830, endOffset=13831), StreamOffsetRange(streamId=208, startOffset=13795, endOffset=13797), StreamOffsetRange(streamId=209, startOffset=13787, endOffset=13791), StreamOffsetRange(streamId=212, startOffset=13800, endOffset=13802), StreamOffsetRange(streamId=213, startOffset=13803, endOffset=13805), StreamOffsetRange(streamId=214, startOffset=13792, endOffset=13793), StreamOffsetRange(streamId=215, startOffset=13793, endOffset=13794), StreamOffsetRange(streamId=238, startOffset=13787, endOffset=13789), StreamOffsetRange(streamId=264, startOffset=3107112, endOffset=3107340), StreamOffsetRange(streamId=287, startOffset=3098976, endOffset=3099192), StreamOffsetRange(streamId=312, startOffset=3099528, endOffset=3099756), StreamOffsetRange(streamId=313, startOffset=3101952, endOffset=3102156), StreamOffsetRange(streamId=316, startOffset=3103128, endOffset=3103416), StreamOffsetRange(streamId=326, startOffset=3099864, endOffset=3100092), StreamOffsetRange(streamId=335, startOffset=3102780, endOffset=3102996), StreamOffsetRange(streamId=336, startOffset=3103632, endOffset=3103860), StreamOffsetRange(streamId=337, startOffset=3102708, endOffset=3102948), StreamOffsetRange(streamId=338, startOffset=3100116, endOffset=3100344), StreamOffsetRange(streamId=339, startOffset=3100284, endOffset=3100512), StreamOffsetRange(streamId=340, startOffset=3103548, endOffset=3103764), StreamOffsetRange(streamId=357, startOffset=3103572, endOffset=3103788), StreamOffsetRange(streamId=361, startOffset=3100944, endOffset=3101172), StreamOffsetRange(streamId=63, startOffset=18858143, endOffset=18859456), StreamOffsetRange(streamId=66, startOffset=18843408, endOffset=18844838), StreamOffsetRange(streamId=94, startOffset=18859908, endOffset=18861204), StreamOffsetRange(streamId=69, startOffset=18849607, endOffset=18850986), StreamOffsetRange(streamId=95, startOffset=18847409, endOffset=18848724), StreamOffsetRange(streamId=98, startOffset=18845086, endOffset=18846344), StreamOffsetRange(streamId=99, startOffset=18857041, endOffset=18858426), StreamOffsetRange(streamId=97, startOffset=18845226, endOffset=18846573), StreamOffsetRange(streamId=100, startOffset=18841493, endOffset=18842878), StreamOffsetRange(streamId=101, startOffset=18838950, endOffset=18840305), StreamOffsetRange(streamId=110, startOffset=18854824, endOffset=18856315), StreamOffsetRange(streamId=119, startOffset=18873372, endOffset=18874755), StreamOffsetRange(streamId=112, startOffset=18845500, endOffset=18846813), StreamOffsetRange(streamId=246, startOffset=18881433, endOffset=18882810), StreamOffsetRange(streamId=248, startOffset=18827836, endOffset=18829190), StreamOffsetRange(streamId=249, startOffset=18848284, endOffset=18849576), StreamOffsetRange(streamId=261, startOffset=18834267, endOffset=18835667), StreamOffsetRange(streamId=253, startOffset=18833837, endOffset=18835234), StreamOffsetRange(streamId=265, startOffset=18857910, endOffset=18859300), StreamOffsetRange(streamId=268, startOffset=18837165, endOffset=18838541), StreamOffsetRange(streamId=271, startOffset=18853192, endOffset=18854545), StreamOffsetRange(streamId=270, startOffset=18832904, endOffset=18834333), StreamOffsetRange(streamId=269, startOffset=18856700, endOffset=18858500), StreamOffsetRange(streamId=272, startOffset=18860625, endOffset=18861937), StreamOffsetRange(streamId=289, startOffset=18865377, endOffset=18866740), StreamOffsetRange(streamId=299, startOffset=18839288, endOffset=18840706), StreamOffsetRange(streamId=266, startOffset=18851296, endOffset=18852765)] advance check failed, error: OFFSET_NOT_MATCHED (org.apache.kafka.controller.stream.StreamControlManager)
```
We can see that these two objects' contents are the same, indicating they are probably the same object.

Also in node-103's server.log:
```
[2024-03-29 17:09:03,431] ERROR Error while committing stream set object: CommitStreamSetObjectRequestData(nodeId=103, nodeEpoch=1711605353154, objectId=1292168, orderId=1292168, objectSize=101567, objectStreamRanges=[ObjectStreamRange(streamId=3, streamEpoch=0, startOffset=13802, endOffset=13804), ObjectStreamRange(streamId=8, streamEpoch=0, startOffset=13807, endOffset=13808), ObjectStreamRange(streamId=9, streamEpoch=0, startOffset=13822, endOffset=13823), ObjectStreamRange(streamId=37, streamEpoch=0, startOffset=13793, endOffset=13794), ObjectStreamRange(streamId=38, streamEpoch=0, startOffset=13780, endOffset=13781), ObjectStreamRange(streamId=39, streamEpoch=0, startOffset=13756, endOffset=13757), ObjectStreamRange(streamId=40, streamEpoch=0, startOffset=13769, endOffset=13770), ObjectStreamRange(streamId=41, streamEpoch=0, startOffset=13794, endOffset=13796), ObjectStreamRange(streamId=42, streamEpoch=0, startOffset=13777, endOffset=13778), ObjectStreamRange(streamId=43, streamEpoch=0, startOffset=13776, endOffset=13777), ObjectStreamRange(streamId=75, streamEpoch=0, startOffset=3103716, endOffset=3103932), ObjectStreamRange(streamId=93, streamEpoch=0, startOffset=3101424, endOffset=3101664), ObjectStreamRange(streamId=129, streamEpoch=0, startOffset=3102384, endOffset=3102612), ObjectStreamRange(streamId=136, streamEpoch=0, startOffset=3101160, endOffset=3101364), ObjectStreamRange(streamId=140, streamEpoch=0, startOffset=3106608, endOffset=3106824), ObjectStreamRange(streamId=148, streamEpoch=0, startOffset=3100716, endOffset=3100932), ObjectStreamRange(streamId=149, streamEpoch=0, startOffset=3101244, endOffset=3101460), ObjectStreamRange(streamId=150, streamEpoch=0, startOffset=3104148, endOffset=3104364), ObjectStreamRange(streamId=151, streamEpoch=0, startOffset=3101796, endOffset=3102012), ObjectStreamRange(streamId=153, streamEpoch=0, startOffset=3102816, endOffset=3103032), ObjectStreamRange(streamId=154, streamEpoch=0, startOffset=3101532, endOffset=3101760), ObjectStreamRange(streamId=170, streamEpoch=0, startOffset=3102456, endOffset=3102696), ObjectStreamRange(streamId=179, streamEpoch=0, startOffset=3101772, endOffset=3101976), ObjectStreamRange(streamId=186, streamEpoch=0, startOffset=13810, endOffset=13812), ObjectStreamRange(streamId=187, streamEpoch=0, startOffset=13804, endOffset=13806), ObjectStreamRange(streamId=189, streamEpoch=0, startOffset=13814, endOffset=13815), ObjectStreamRange(streamId=190, streamEpoch=0, startOffset=13830, endOffset=13831), ObjectStreamRange(streamId=208, streamEpoch=0, startOffset=13795, endOffset=13797), ObjectStreamRange(streamId=209, streamEpoch=0, startOffset=13787, endOffset=13791), ObjectStreamRange(streamId=212, streamEpoch=0, startOffset=13800, endOffset=13802), ObjectStreamRange(streamId=213, streamEpoch=0, startOffset=13803, endOffset=13805), ObjectStreamRange(streamId=214, streamEpoch=0, startOffset=13792, endOffset=13793), ObjectStreamRange(streamId=215, streamEpoch=0, startOffset=13793, endOffset=13794), ObjectStreamRange(streamId=238, streamEpoch=0, startOffset=13787, endOffset=13789), ObjectStreamRange(streamId=264, streamEpoch=0, startOffset=3107112, endOffset=3107340), ObjectStreamRange(streamId=287, streamEpoch=0, startOffset=3098976, endOffset=3099192), ObjectStreamRange(streamId=312, streamEpoch=0, startOffset=3099528, endOffset=3099756), ObjectStreamRange(streamId=313, streamEpoch=0, startOffset=3101952, endOffset=3102156), ObjectStreamRange(streamId=316, streamEpoch=0, startOffset=3103128, endOffset=3103416), ObjectStreamRange(streamId=326, streamEpoch=0, startOffset=3099864, endOffset=3100092), ObjectStreamRange(streamId=335, streamEpoch=0, startOffset=3102780, endOffset=3102996), ObjectStreamRange(streamId=336, streamEpoch=0, startOffset=3103632, endOffset=3103860), ObjectStreamRange(streamId=337, streamEpoch=0, startOffset=3102708, endOffset=3102948), ObjectStreamRange(streamId=338, streamEpoch=0, startOffset=3100116, endOffset=3100344), ObjectStreamRange(streamId=339, streamEpoch=0, startOffset=3100284, endOffset=3100512), ObjectStreamRange(streamId=340, streamEpoch=0, startOffset=3103548, endOffset=3103764), ObjectStreamRange(streamId=357, streamEpoch=0, startOffset=3103572, endOffset=3103788), ObjectStreamRange(streamId=361, streamEpoch=0, startOffset=3100944, endOffset=3101172)], streamObjects=[StreamObject(objectId=1292169, objectSize=18845369, streamId=63, startOffset=18858143, endOffset=18859456), StreamObject(objectId=1292170, objectSize=20605421, streamId=66, startOffset=18843408, endOffset=18844838), StreamObject(objectId=1292172, objectSize=18452663, streamId=94, startOffset=18859908, endOffset=18861204), StreamObject(objectId=1292171, objectSize=19825431, streamId=69, startOffset=18849607, endOffset=18850986), StreamObject(objectId=1292173, objectSize=18827245, streamId=95, startOffset=18847409, endOffset=18848724), StreamObject(objectId=1292175, objectSize=18253912, streamId=98, startOffset=18845086, endOffset=18846344), StreamObject(objectId=1292176, objectSize=19987907, streamId=99, startOffset=18857041, endOffset=18858426), StreamObject(objectId=1292174, objectSize=19395194, streamId=97, startOffset=18845226, endOffset=18846573), StreamObject(objectId=1292177, objectSize=19926531, streamId=100, startOffset=18841493, endOffset=18842878), StreamObject(objectId=1292178, objectSize=19453587, streamId=101, startOffset=18838950, endOffset=18840305), StreamObject(objectId=1292179, objectSize=21461007, streamId=110, startOffset=18854824, endOffset=18856315), StreamObject(objectId=1292181, objectSize=19707925, streamId=119, startOffset=18873372, endOffset=18874755), StreamObject(objectId=1292180, objectSize=18869991, streamId=112, startOffset=18845500, endOffset=18846813), StreamObject(objectId=1292182, objectSize=19830046, streamId=246, startOffset=18881433, endOffset=18882810), StreamObject(objectId=1292183, objectSize=19104665, streamId=248, startOffset=18827836, endOffset=18829190), StreamObject(objectId=1292184, objectSize=18695700, streamId=249, startOffset=18848284, endOffset=18849576), StreamObject(objectId=1292186, objectSize=20067807, streamId=261, startOffset=18834267, endOffset=18835667), StreamObject(objectId=1292185, objectSize=19871556, streamId=253, startOffset=18833837, endOffset=18835234), StreamObject(objectId=1292187, objectSize=19716157, streamId=265, startOffset=18857910, endOffset=18859300), StreamObject(objectId=1292189, objectSize=19934410, streamId=268, startOffset=18837165, endOffset=18838541), StreamObject(objectId=1292192, objectSize=19402293, streamId=271, startOffset=18853192, endOffset=18854545), StreamObject(objectId=1292191, objectSize=20500248, streamId=270, startOffset=18832904, endOffset=18834333), StreamObject(objectId=1292190, objectSize=25413288, streamId=269, startOffset=18856700, endOffset=18858500), StreamObject(objectId=1292193, objectSize=18937810, streamId=272, startOffset=18860625, endOffset=18861937), StreamObject(objectId=1292194, objectSize=19596857, streamId=289, startOffset=18865377, endOffset=18866740), StreamObject(objectId=1292195, objectSize=20361519, streamId=299, startOffset=18839288, endOffset=18840706), StreamObject(objectId=1292188, objectSize=21042994, streamId=266, startOffset=18851296, endOffset=18852765)], compactedObjectIds=[], failoverMode=false), code: OFFSET_NOT_MATCHED, retry later (kafka.log.stream.s3.objects.ControllerObjectManager)
...
```
It's node 103 that commit the object 1292168 multiple times.

In `StreamControlManager#commitStreamSetObject` we can see that the controller check the continuity before the redundancy:
![image](https://github.com/AutoMQ/automq/assets/78631860/0b25e02e-2d60-48f5-aedb-8d6134b9d199)

What happened:
1. Node 103 sent commit object request with objectId=1292168 to the controller.
2. The pressure on the controller was too high, so it failed to process this request in a timely manner (but still finished it later).
3. Node 103 considered this request to have timed out and suggested waiting for a while before retrying.
4. The controller checked the new request and returned OFFSET_NOT_MATCHED to Node 103.

In summary, this issue is caused by the non-idempotency of the `commitStreamSetObject` on the Controller.
